testing/integration/test.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
```

Comment:
Use a separate test script that can be run many times. At the end of a test with large JWTs and caching disabled, the response.txt file shows how data has been read past the end of a buffer:

"headers": {
    "Accept": "*/*", 
    "Authorization": "Bearer eyJraWQiOiItMjAyMDIxMjM3MyIsIng1dCI6IktwdlhqdEJQeWx3RHNENWtWLTN1bkVzQXozRSIsImFsZyI6IlJTMjU2In0.eyJqdGkiOiIwMjNkYzliYi0zOGFjLTQ3ZGMtOTk3My01ODliZmM2MWNmMmUiLCJkZWxlZ2F0aW9uSWQiOiIyZDVkYzA1MC0xYTI0LTQyZWYtYTkyNy00MTFkODVjYzJjYTEiLCJleHAiOjE3MzEzNDQzMTcsIm5iZiI6MTczMTM0NDAxNywic2NvcGUiOiJleGFtcGxlIiwiaXNzIjoiaHR0cDovL2lkZW50aXR5c2VydmVyOjg0NDMvb2F1dGgvdjIvb2F1dGgtYW5vbnltb3VzIiwic3ViIjoidGVzdC1jbGllbnQiLCJhdWQiOiJ0ZXN0LWNsaWVudCIsImlhdCI6MTczMTM0NDAxNywicHVycG9zZSI6ImFjY2Vzc190b2tlbiIsImV4YW1wbGUiOiJMb3JlbSBpcHN1bSBkb2xvciBzaXQgYW1ldCwgY29uc2VjdGV0dXIgYWRpcGlzY2luZyBlbGl0LCBzZWQgZG8gZWl1c21vZCB0ZW1wb3IgaW5jaWRpZHVudCB1dCBsYWJvcmUgZXQgZG9sb3JlIG1hZ25hIGFsaXF1YS4gVXQgZW5pbSBhZCBtaW5pbSB2ZW5pYW0sIHF1aXMgbm9zdHJ1ZCBleGVyY2l0YXRpb24gdWxsYW1jbyBsYWJvcmlzIG5pc2kgdXQgYWxpcXVpcCBleCBlYSBjb21tb2RvIGNvbnNlcXVhdC4gRHVpcyBhdXRlIGlydXJlIGRvbG9yIGluIHJlcHJlaGVuZGVyaXQgaW4gdm9sdXB0YXRlIHZlbGl0IGVzc2UgY2lsbHVtIGRvbG9yZSBldSBmdWdpYXQgbnVsbGEgcGFyaWF0dXIuIEV4Y2VwdGV1ciBzaW50IG9jY2FlY2F0IGN1cGlkYXRhdCBub24gcHJvaWRlbnQsIHN1bnQgaW4gY3VscGEgcXVpIG9mZmljaWEgZGVzZXJ1bnQgbW9sbGl0IGFuaW0gaWQgZXN0IGxhYm9ydW0gU2VkIHV0IHBlcnNwaWNpYXRpcyB1bmRlIG9tbmlzIGlzdGUgbmF0dXMgZXJyb3Igc2l0IHZvbHVwdGF0ZW0gYWNjdXNhbnRpdW0gZG9sb3JlbXF1ZSBsYXVkYW50aXVtLCB0b3RhbSByZW0gYXBlcmlhbSwgZWFxdWUgaXBzYSBxdWFlIGFiIGlsbG8gaW52ZW50b3JlIHZlcml0YXRpcyBldCBxdWFzaSBhcmNoaXRlY3RvIGJlYXRhZSB2aXRhZSBkaWN0YSBzdW50IGV4cGxpY2Fiby4gTmVtbyBlbmltIGlwc2FtIHZvbHVwdGF0ZW0gcXVpYSB2b2x1cHRhcyBzaXQgYXNwZXJuYXR1ciBhdXQgb2RpdCBhdXQgZnVnaXQsIHNlZCBxdWlhIGNvbnNlcXV1bnR1ciBtYWduaSBkb2xvcmVzIGVvcyBxdWkgcmF0aW9uZSB2b2x1cHRhdGVtIHNlcXVpIG5lc2NpdW50LiBOZXF1ZSBwb3JybyBxdWlzcXVhbSBlc3QsIHF1aSBkb2xvcmVtIGlwc3VtIHF1aWEgZG9sb3Igc2l0IGFtZXQsIGNvbnNlY3RldHVyLCBhZGlwaXNjaSB2ZWxpdCwgc2VkIHF1aWEgbm9uIG51bXF1YW0gZWl1cyBtb2RpIHRlbXBvcmEgaW5jaWR1bnQgdXQgbGFib3JlIGV0IGRvbG9yZSBtYWduYW0gYWxpcXVhbSBxdWFlcmF0IHZvbHVwdGF0ZW0uIFV0IGVuaW0gYWQgbWluaW1hIHZlbmlhbSwgcXVpcyBub3N0cnVtIGV4ZXJjaXRhdGlvbmVtIHVsbGFtIGNvcnBvcmlzIHN1c2NpcGl0IGxhYm9yaW9zYW0sIG5pc2kgdXQgYWxpcXVpZCBleCBlYSBjb21tb2RpIGNvbnNlcXVhdHVyPyBRdWlzIGF1dGVtIHZlbCBldW0gaXVyZSByZXByZWhlbmRlcml0IHF1aSBpbiBlYSB2b2x1cHRhdGUgdmVsaXQgZXNzZSBxdWFtIG5paGlsIG1vbGVzdGlhZSBjb25zZXF1YXR1ciwgdmVsIGlsbHVtIHF1aSBkb2xvcmVtIGV1bSBmdWdpYXQgcXVvIHZvbHVwdGFzIG51bGxhIHBhcmlhdHVyIEJ1dCBJIG11c3QgZXhwbGFpbiB0byB5b3UgaG93IGFsbCB0aGlzIG1pc3Rha2VuIGlkZWEgb2YgZGVub3VuY2luZyBwbGVhc3VyZSBhbmQgcHJhaXNpbmcgcGFpbiB3YXMgYm9ybiBhbmQgSSB3aWxsIGdpdmUgeW91IGEgY29tcGxldGUgYWNjb3VudCBvZiB0aGUgc3lzdGVtLCBhbmQgZXhwb3VuZCB0aGUgYWN0dWFsIHRlYWNoaW5ncyBvZiB0aGUgZ3JlYXQgZXhwbG9yZXIgb2YgdGhlIHRydXRoLCB0aGUgbWFzdGVyLWJ1aWxkZXIgb2YgaHVtYW4gaGFwcGluZXNzLiBObyBvbmUgcmVqZWN0cywgZGlzbGlrZXMsIG9yIGF2b2lkcyBwbGVhc3VyZSBpdHNlbGYsIGJlY2F1c2UgaXQgaXMgcGxlYXN1cmUsIGJ1dCBiZWNhdXNlIHRob3NlIHdobyBkbyBub3Qga25vdyBob3cgdG8gcHVyc3VlIHBsZWFzdXJlIHJhdGlvbmFsbHkgZW5jb3VudGVyIGNvbnNlcXVlbmNlcyB0aGF0IGFyZSBleHRyZW1lbHkgcGFpbmZ1bC4gTm9yIGFnYWluIGlzIHRoZXJlIGFueW9uZSB3aG8gbG92ZXMgb3IgcHVyc3VlcyBvciBkZXNpcmVzIHRvIG9idGFpbiBwYWluIG9mIGl0c2VsZiwgYmVjYXVzZSBpdCBpcyBwYWluLCBidXQgYmVjYXVzZSBvY2Nhc2lvbmFsbHkgY2lyY3Vtc3RhbmNlcyBvY2N1ciBpbiB3aGljaCB0b2lsIGFuZCBwYWluIGNhbiBwcm9jdXJlIGhpbSBzb21lIGdyZWF0IHBsZWFzdXJlLiBUbyB0YWtlIGEgdHJpdmlhbCBleGFtcGxlLCB3aGljaCBvZiB1cyBldmVyIHVuZGVydGFrZXMgbGFib3Jpb3VzIHBoeXNpY2FsIGV4ZXJjaXNlLCBleGNlcHQgdG8gb2J0YWluIHNvbWUgYWR2YW50YWdlIGZyb20gaXQ_IEJ1dCB3aG8gaGFzIGFueSByaWdodCB0byBmaW5kIGZhdWx0IHdpdGggYSBtYW4gd2hvIGNob29zZXMgdG8gZW5qb3kgYSBwbGVhc3VyZSB0aGF0IGhhcyBubyBhbm5veWluZyBjb25zZXF1ZW5jZXMsIG9yIG9uZSB3aG8gYXZvaWRzIGEgcGFpbiB0aGF0IHByb2R1Y2VzIG5vIHJlc3VsdGFudCBwbGVhc3VyZUF0IHZlcm8gZW9zIGV0IGFjY3VzYW11cyBldCBpdXN0byBvZGlvIGRpZ25pc3NpbW9zIGR1Y2ltdXMgcXVpIGJsYW5kaXRpaXMgcHJhZXNlbnRpdW0gdm9sdXB0YXR1bSBkZWxlbml0aSBhdHF1ZSBjb3JydXB0aSBxdW9zIGRvbG9yZXMgZXQgcXVhcyBtb2xlc3RpYXMgZXhjZXB0dXJpIHNpbnQg\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0019\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000Bearer \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0